server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -664,6 +667,60 @@ public void testCacheIndexFilesClearedOnDelete() throws Exception {
         logger.info("--> validated that the cache file path doesn't exist");
     }
 
+    /**
+     * Test scenario that validates that the default search preference for searchable snapshot
+     * is primary shards
+     */
+    public void testDefaultShardPreference() throws Exception {
+        // Increase the probability of hitting a replica
+        final int numReplicas = randomIntBetween(5, 10);
```

Comment:
What is non-deterministic about this test?

Reply:
I had increased the number of replicas to hit in case of search with a non primary preference. I did find an alternative with the `ClusterSearchShards` API which returns possible options instead of the selected shard only. 

The number of replicas will not cause in the test determinism with the current API used within the test. (It is deterministic with the `ClusterSearchShards` API)

Reply:
Would it be okay to set this to a fixed value, like 1? 10 replicas makes the test pretty heavyweight, and it is good to make these things as lean as possible given the problems we've seen with flaky failures.

Reply:
Updated the test to be leaner.